lib/book.rb

```diff
@@ -0,0 +1,12 @@
+class Book
+  attr_reader :title, :author, :publication_year, :publication_date
+
+  def initialize(attributes)
+    @title = attributes[:title]
+    first_name = attributes[:author_first_name]
+    last_name = attributes[:author_last_name]
+    @author = "#{first_name} #{last_name}"
```

Comment:
You could also interpolate from the attributes directly (without creating variables)

`@author = "#{attributes[:author_first_name]} #{attributes[:author_last_name]}"`